wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -1297,6 +1294,52 @@ public List<DFULogicalFileWrapper> searchFiles(String logicalFilename, String cl
         return result;
     }
 
+
+    /**
+     * searchFiles
+     *
+     * @param request - the search request
+     * @return - collection of files matching the request passed in
+     * @throws java.lang.Exception general exception
+     * @throws org.hpccsystems.ws.client.wrappers.ArrayOfEspExceptionWrapper array of esp exception wrapper
+     */
+    public List<DFULogicalFileWrapper> searchFiles(DFUQueryRequestWrapper request) throws Exception, ArrayOfEspExceptionWrapper
+    {
+        verifyStub(); // Throws exception if stub failed
+
+        if (request.getLogicalName() != null && request.getLogicalName().startsWith("~"))
```

Comment:
If there's ever a situation where the leading tilde is necessary, we should add an input param to this method to allow caller to include the tilde
Also, since we are changing the caller's input, we should report this to the logger

Reply:
this is being done in all the search methods--if we change it here we should also change it in the other search methods. Separate PR?

Reply:
Gotcha. It feels like the previous method should populate a DFUQueryRequestWrapper and call into yours (to avoid replicated logic). We can handle that as an improvement in a separate PR. https://track.hpccsystems.com/browse/JAPI-460